include/eve/module/real/core/function/regular/simd/x86/average.hpp

```diff
@@ -16,32 +16,24 @@
 
 namespace eve::detail
 {
-  // -----------------------------------------------------------------------------------------------
-  // 128 bits implementation
   template<unsigned_scalar_value T, typename N>
-  EVE_FORCEINLINE wide<T, N, x86_128_> average_(EVE_SUPPORTS(sse2_)
-                                           , wide<T, N, x86_128_> const &a
-                                           , wide<T, N, x86_128_> const &b) noexcept
+  EVE_FORCEINLINE wide<T, N> average_(EVE_SUPPORTS(sse2_), wide<T, N> a, wide<T, N> b) noexcept
+    requires x86_abi<abi_t<T, N>>
   {
-         if constexpr(sizeof(T) == 1) return _mm_avg_epu8(a, b);
-    else if constexpr(sizeof(T) == 2) return _mm_avg_epu16(a, b);
-    else                              return (a & b) + ((a ^ b) >> 1);
-  }
-
-  // -----------------------------------------------------------------------------------------------
-  // 256 bits implementation
-  template<unsigned_scalar_value T, typename N>
-  EVE_FORCEINLINE wide<T, N, x86_256_> average_(EVE_SUPPORTS(avx_)
-                                           , wide<T, N, x86_256_> const &a
-                                           , wide<T, N, x86_256_> const &b) noexcept
-  {
-    if constexpr(current_api >= avx2)
+    if constexpr ( sizeof(T) == 2 )
     {
-             if constexpr(sizeof(T) == 1) return _mm256_avg_epu8(a, b);
-        else if constexpr(sizeof(T) == 2) return _mm256_avg_epu16(a, b);
-        else                              return (a & b) + ((a ^ b) >> 1);
```

Comment:
where does this one went ?

Reply:
CPU should  do this https://github.com/jfalcou/eve/blob/1e865bfb459ea3dfb4a6c9b10944c55ae761a841/include/eve/module/real/core/function/regular/generic/average.hpp#L42